py-polars/src/apply/mod.rs

```diff
@@ -54,6 +54,10 @@ fn iterator_to_struct<'a>(
     // ]
     let mut struct_fields: BTreeMap<&str, Vec<AnyValue>> = BTreeMap::new();
 
+    // as a BTreeMap sorts its keys, we also need to track the original
```

Comment:
We can use `PlIndexMap` if insertion order needs to be maintained.

Reply:
Ah, thanks @ritchie46 - my searching did find https://doc.rust-lang.org/stable/nightly-rustc/rustc_data_structures/sorted_map/struct.SortedIndexMultiMap.html but not `PlIndexMap`

Trying `PlIndexMap` gives me this back from the compiler, I'm not sure how to proceed.
```rust
error[E0599]: the method `par_iter` exists for struct `IndexMap<&str, Vec<AnyValue<'_>>, RandomState>`, but its trait bounds were not satisfied
   --> src/apply/mod.rs:114:14
    |
113 | /             struct_fields
114 | |             .par_iter()
    | |_____________-^^^^^^^^
    |
   ::: /Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/indexmap-2.0.0/src/map.rs:81:1
    |
81  |   pub struct IndexMap<K, V, S = RandomState> {
    |   ------------------------------------------ doesn't satisfy `_: IntoParallelRefIterator<'_>`
    |
    = note: the following trait bounds were not satisfied:
            `&indexmap::map::IndexMap<&str, Vec<polars_rs::prelude::AnyValue<'_>>, ahash::RandomState>: polars_rs::export::rayon::iter::IntoParallelIterator`
            which is required by `indexmap::map::IndexMap<&str, Vec<polars_rs::prelude::AnyValue<'_>>, ahash::RandomState>: polars_rs::export::rayon::iter::IntoParallelRefIterator<'_>`
```

Reply:
Ah right. Then this solution is fine. Thank you! :+1: